src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
> In other words - using NUMBER(3,0) will be a way of telling kaka-connect-jdbc to map the column to int8.

I think we should err on the side of the wider type for now rather than add potentially unsafe conversions, and use the mappings I suggested in https://github.com/confluentinc/kafka-connect-jdbc/pull/104/files#r72742317.

One option if you need a specific type that the automatic conversions can't address is to implement a custom `Converter` that wraps one of the existing ones and do the down-casting in there. You'd want to hook into the `fromConnectData` since this is a source connector and that method will be called along the way from Connect -> Kafka. [ In future this may become more flexible as we design and integrate 'single-message transforms', e.g. there could be a transformation that allows you to project to a specific Avro schema you define even if it involves down-casting. ]